app/login.go

```diff
@@ -35,6 +53,13 @@ func (a *App) AuthenticateUserForLogin(id, loginId, password, mfaToken string, l
 		return nil, err
 	}
 
+	// If client side cert is enable and it's checking as a primary source
+	// then trust the proxy and cert that the correct user is supplied and allow
+	// them access
+	if *a.Config().ExperimentalSettings.ClientSideCertEnable && *a.Config().ExperimentalSettings.ClientSideCertCheck == model.CLIENT_SIDE_CERT_CHECK_PRIMARY_AUTH {
```

Comment:
Can I get some materials to read on how this is supposed to work. We don't appear to be validating this at all. How do I know the email matches the cert provided? Can't I just login as anyone?

Reply:
Resolved with explanation here: https://pre-release.mattermost.com/core/pl/fk9omytpopgm5qt5r5s1r14cfc